astropy_helpers/commands/build_sphinx.py

```diff
@@ -162,51 +152,16 @@ def run(self):
             for egg in glob.glob(os.path.join(eggs_path, '*.egg')):
                 subproccode += 'sys.path.append({egg!r})\n'.format(egg=egg)
 
-        # runlines[1:] removes 'def run(self)' on the first line
-        subproccode += textwrap.dedent(''.join(runlines[1:]))
-
-        # All "self.foo" in the subprocess code needs to be replaced by the
-        # values taken from the current self in *this* process
-        subproccode = self._self_iden_rex.split(subproccode)
-        for i in range(1, len(subproccode), 2):
-            iden = subproccode[i]
-            val = getattr(self, iden)
-            if iden.endswith('_dir'):
-                # Directories should be absolute, because the `chdir` call
-                # in the new process moves to a different directory
-                subproccode[i] = repr(os.path.abspath(val))
-            else:
-                subproccode[i] = repr(val)
-        subproccode = ''.join(subproccode)
-
-        optcode = textwrap.dedent("""
-
-        class Namespace(object): pass
-        self = Namespace()
-        self.pdb = {pdb!r}
-        self.verbosity = {verbosity!r}
-        self.traceback = {traceback!r}
-
-        """).format(pdb=getattr(self, 'pdb', False),
-                    verbosity=getattr(self, 'verbosity', 0),
-                    traceback=getattr(self, 'traceback', False))
-
-        subproccode = optcode + subproccode
-
-        # This is a quick gross hack, but it ensures that the code grabbed from
-        # SphinxBuildDoc.run will work in Python 2 if it uses the print
-        # function
-        if minversion(sphinx, '1.3'):
-            subproccode = 'from __future__ import print_function' + subproccode
-
-        if self.no_intersphinx:
-            # the confoverrides variable in sphinx.setup_command.BuildDoc can
-            # be used to override the conf.py ... but this could well break
-            # if future versions of sphinx change the internals of BuildDoc,
-            # so remain vigilant!
-            subproccode = subproccode.replace(
-                'confoverrides = {}',
-                'confoverrides = {\'intersphinx_mapping\':{}}')
+        subproccode += textwrap.dedent("""
+
+            if SPHINX_LT_17:
+                argv.insert(0, 'sphinx-build')
+
+            for builder in {builders!r}:
+                argv = ['-W', '.', '-b', builder, os.path.join({output_dir!r}, builder)]
+                build_main(argv=argv)
+        """.format(builders=self.builder,
+                   output_dir=os.path.relpath(self.build_dir, 'docs')))
```

Comment:
Better but not quite there as "docs" is still hardcoded. There are projects with say, "doc", instead of "docs".

Reply:
I think that should be fixed now